src/EditorFeatures/CSharpTest/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3989,6 +3990,33 @@ static void Main()
 }");
         }
 
+        [WorkItem(29264, "https://github.com/dotnet/roslyn/issues/29264")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public async Task DontRemoveCastOnDictionaryIndexer()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+
+static class Program
+{
+    enum TestEnum
+    {
+        Test,
+    }
+
+    static void Main()
+    {
+        Dictionary<int, string> Icons = new Dictionary<int, string>
+        {
+            [[|(int)|] TestEnum.Test] = null,
+        };
+    }
+}");
+        }
```

Comment:
can you have a test where this isn't passing an enum, but is passing an integer?  we should ensure we do remove the case there.